### PR TITLE
(CDAP-960) Added batch endpoint for stream

### PR DESCRIPTION
**Pull request by chtyim**
- Added support for line separated text content

### CODE REVIEW

File: cdap-data-fabric/src/main/java/co/cask/cdap/data/stream/service/upload/FileContentWriter.java [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+    this.streamWriter = streamWriter;
+    this.streamEventData = new MutableStreamEventData();
+    this.streamEvent = new MutableStreamEvent();
+
+    directory.mkdirs();
+    this.eventFile = directory.append(String.format("upload.%s.dat", streamConfig.getName()));
+    this.indexFile = directory.append(String.format("upload.%s.idx", streamConfig.getName()));
+
+    Map<String, String> properties = createStreamFileProperties(headers);
+    properties.put(StreamDataFileConstants.Property.Key.UNI_TIMESTAMP,
```
Comment (nitinmotgi) [inline cdap-data-fabric/src/main/java/co/cask/cdap/data/stream/service/upload/FileContentWriter.java]: UNI_TIMESTAMP is that a typo ?

File: cdap-data-fabric/src/main/java/co/cask/cdap/data/stream/service/upload/FileContentWriterFactory.java [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    this.streamConfig = streamConfig;
+    this.streamWriter = streamWriter;
+    this.headers = ImmutableMap.copyOf(headers);
+
+    Location location = streamConfig.getLocation();
+    if (location == null) {
+      // Should never happen
+      throw new IllegalArgumentException("Unknown stream location for stream " + streamConfig.getName());
+    }
+    this.uploadDirectory = location.append("tmp")
```
Comment (nitinmotgi) [inline cdap-data-fabric/src/main/java/co/cask/cdap/data/stream/service/upload/FileContentWriterFactory.java]: @chtyim does the temporary dir get deleted or renamed ?

**Comment by nitinmotgi** [comment]
Minor comments otherwise this look good and it's very clear :+1:

File: cdap-data-fabric/src/main/java/co/cask/cdap/data/stream/service/upload/FileContentWriter.java (hunk 1 above)
Comment (chtyim) [inline cdap-data-fabric/src/main/java/co/cask/cdap/data/stream/service/upload/FileContentWriter.java]: No. The constant is called `UNI_TIMESTAMP`. It's a special property saying all events in the stream file has the same timestamp.

File: cdap-data-fabric/src/main/java/co/cask/cdap/data/stream/service/upload/FileContentWriterFactory.java (hunk 2 above)
Comment (chtyim) [inline cdap-data-fabric/src/main/java/co/cask/cdap/data/stream/service/upload/FileContentWriterFactory.java]: There is cleanup inside the ContentWriter. However, I found that might be accidentally remove all files that are still uploading. Updated the code. Please look at it again.

**Comment by gokulavasan** [comment]
Additional commit LGTM :+1:

**Pull request merged**